src/main/java/com/google/sps/servlets/IntervieweeFeedbackServlet.java

```diff
@@ -0,0 +1,133 @@
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+
+// Servlet that gets the feedback from an interviwer and sends it to an interviewee.
+@WebServlet("/interviewee-feedback")
+public class IntervieweeFeedbackServlet extends HttpServlet {
+  private ScheduledInterviewDao scheduledInterviewDao;
+  private PersonDao personDao;
+  private Path emailsPath =
+      Paths.get(
+          System.getProperty("user.home") + "/InterviewMe/src/main/resources/templates/email");
+
+  @Override
+  public void init() {
+    init(new DatastoreScheduledInterviewDao(), new DatastorePersonDao());
+  }
+
+  public void init(ScheduledInterviewDao scheduledInterviewDao, PersonDao personDao) {
+    this.scheduledInterviewDao = scheduledInterviewDao;
+    this.personDao = personDao;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long scheduledInterviewId = Long.parseLong(request.getParameter("interviewId"));
+    int numberOfQuestions = Integer.parseInt(request.getParameter("questionCount"));
+    HashMap<String, String> answers = new HashMap<String, String>();
+    for (int i = 1; i <= numberOfQuestions; i++) {
+      String template = String.format("{{question_%s}}", i);
+      String param = String.format("question%s", i);
+      answers.put(template, request.getParameter(param));
+    }
+
+    String userEmail = UserServiceFactory.getUserService().getCurrentUser().getEmail();
+    String userId = UserServiceFactory.getUserService().getCurrentUser().getUserId();
+    // Since Users returned from the LocalUserService (in tests) do not have userIds, here we set
+    // the userId equal to a hashcode.
+    if (userId == null) {
+      userId = String.format("%d", userEmail.hashCode());
+    }
+
+    if (!interviewExists(scheduledInterviewId)) {
+      response.sendError(HttpServletResponse.SC_NOT_FOUND);
+      return;
+    }
+    ScheduledInterview scheduledInterview = getInterview(scheduledInterviewId);
+    answers.put("{{formatted_date}}", scheduledInterview.getDateString());
+    if (isInterviewer(scheduledInterview, userId)) {
+      try {
+        sendFeedback(getIntervieweeEmail(scheduledInterview), answers);
+      } catch (Exception e) {
+        e.printStackTrace();
+        response.sendError(HttpServletResponse.SC_REQUEST_TIMEOUT);
+        return;
+      }
+      response.sendRedirect("/scheduled-interviews.html");
+      return;
+    } else {
+      response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+      return;
+    }
+  }
+
+  private boolean interviewExists(long scheduledInterviewId) {
+    return scheduledInterviewDao.get(scheduledInterviewId).isPresent();
+  }
+
+  private ScheduledInterview getInterview(long scheduledInterviewId) {
+    return scheduledInterviewDao.get(scheduledInterviewId).get();
```

Comment:
Return the optional here, and handle the absence of the value by returning a code (rather than throwing an exception).

Reply:
Done